tracing/tempo/http_client.go

```diff
@@ -50,13 +61,24 @@ func NewOtelClient(client http.Client, baseURL *url.URL) (otelClient *OtelHTTPCl
 		}
 	}
 
-	return &OtelHTTPClient{ClusterTag: tags}, nil
+	otelHTTPClient := &OtelHTTPClient{ClusterTag: tags}
+	if config.Get().ExternalServices.Tracing.TempoConfig.CacheEnabled {
+		vanillaStore := store.New[string, *model.TracingSingleTrace]()
+		fifoStore := store.NewFIFOStore[string, *model.TracingSingleTrace](vanillaStore, config.Get().ExternalServices.Tracing.TempoConfig.CacheCapacity)
+		otelHTTPClient.FIFOStore = fifoStore
+		otelHTTPClient.TempoCache = store.NewExpirationStore[string, *model.TracingSingleTrace](context.Background(), fifoStore, util.AsPtr(TTL), util.AsPtr(expirationCheckInterval))
```

Comment:
It'd be nice to pass the context into `NewOtelClient` and down through here. Otherwise the goroutine that expires keys will never stop.

Reply:
Updated to pass the context